presto-main/src/test/java/io/prestosql/operator/scalar/TestConditions.java

```diff
@@ -442,4 +442,15 @@ public void testSimpleCase()
                 DOUBLE,
                 33.0);
     }
+
+    @Test
+    public void testSimpleCase1()
+    {
+        assertFunction("case 8 " +
+                        "when double '76.1' then double '1.0' " +
+                        "when real '8.0' then null " +
+                        "end",
+                DOUBLE,
+                null);
```

Comment:
Note: the test passes before applying the code changes here.

```
presto:default> SELECT case 8 when double '76.1' then double '1.0' when real '8.0' then null end;
 _col0
-------
  NULL
(1 row)
```

```
presto:default> SELECT typeof(case 8 when double '76.1' then double '1.0' when real '8.0' then null end);
 _col0
--------
 double
(1 row)
```

Reply:
```sql
SELECT CASE 8 when 9 then 1 when cast(null as decimal) then 3 else 4 end 
```

Reply:
Indeed.
```
presto:default> SELECT CASE 8 when 9 then 1 when cast(null as decimal(20,1)) then 3 else 4 end ;

Query 20191022_130929_00006_x3p8n failed: Cannot cast java.lang.Long to io.airlift.slice.Slice
java.lang.ClassCastException: Cannot cast java.lang.Long to io.airlift.slice.Slice
	at java.base/java.lang.Class.cast(Class.java:3605)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:710)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
	at io.prestosql.sql.InterpretedFunctionInvoker.invoke(InterpretedFunctionInvoker.java:88)
	at io.prestosql.sql.planner.ExpressionInterpreter$Visitor.invokeOperator(ExpressionInterpreter.java:1298)
	at io.prestosql.sql.planner.ExpressionInterpreter$Visitor.isEqual(ExpressionInterpreter.java:515)
	at io.prestosql.sql.planner.ExpressionInterpreter$Visitor.visitSimpleCaseExpression(ExpressionInterpreter.java:498)
	at io.prestosql.sql.tree.SimpleCaseExpression.accept(SimpleCaseExpression.java:70)
	at io.prestosql.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.prestosql.sql.planner.ExpressionInterpreter.optimize(ExpressionInterpreter.java:300)
	at io.prestosql.sql.planner.iterative.rule.SimplifyExpressions.rewrite(SimplifyExpressions.java:53)
	at io.prestosql.sql.planner.iterative.rule.SimplifyExpressions.lambda$createRewrite$0(SimplifyExpressions.java:78)
	at io.prestosql.sql.planner.iterative.rule.ExpressionRewriteRuleSet$ProjectExpressionRewrite.lambda$apply$0(ExpressionRewriteRuleSet.java:125)
	at io.prestosql.sql.planner.plan.Assignments.lambda$rewrite$1(Assignments.java:108)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at io.prestosql.sql.planner.plan.Assignments.rewrite(Assignments.java:109)
	at io.prestosql.sql.planner.iterative.rule.ExpressionRewriteRuleSet$ProjectExpressionRewrite.apply(ExpressionRewriteRuleSet.java:125)
	at io.prestosql.sql.planner.iterative.rule.ExpressionRewriteRuleSet$ProjectExpressionRewrite.apply(ExpressionRewriteRuleSet.java:106)
	at io.prestosql.sql.planner.iterative.IterativeOptimizer.transform(IterativeOptimizer.java:165)
...
```

Reply:
Thanks. Can you add that case to the test, instead?